code_challenges/code_wars/popping_balloons.py

```diff
@@ -0,0 +1,77 @@
+class Balloons:
+    def __init__(self, balloons):
+        self.balloons = balloons
+        self.balloons_amount = {}
+        self.popped_ballons = []
+
+    
+    def track_balloons(self):
+        for balloon in self.balloons:
+            self.balloons_amount[balloon] = self.balloons_amount.get(balloon, 0) + 1
+        return self.balloons_amount
+
+    def find_top_balloons(self):
+        top_amount = 0
+        second_amount = 0
+        top_balloon = None
+        second_balloon = None
+        
+        for balloon in self.balloons_amount:
+            if self.balloons_amount[balloon] > top_amount:
+                second_amount = top_amount
+                second_balloon = top_balloon
+                top_amount = self.balloons_amount[balloon]
+                top_balloon = balloon
+            elif self.balloons_amount[balloon] > second_amount and self.balloons_amount[balloon] <= top_amount:
+                second_amount = self.balloons_amount[balloon]
+                second_balloon = balloon
+        return top_balloon, top_amount, second_balloon, second_amount
+
+    def pop_most(self, amount):
+        
+        for _ in range(amount):
+            top_balloon, top_amount, second_balloon, second_amount = self.find_top_balloons()
+            one_left = all(x == 1 for x in self.balloons_amount.values())
+            if one_left:
+                self.popped_ballons.append(self.balloons.pop(-1))
+                continue
+            
+            if top_amount > second_amount:
+                    self.balloons_amount[top_balloon] -= 1
+                    for i in range(len(self.balloons) -1, -1, -1):
+                        if self.balloons[i] == top_balloon:
+                            self.popped_ballons.append(self.balloons[i])
+                            self.balloons.pop(i)
+                            break
+            elif top_amount == second_amount:
+                the_biggest = None
+                for i in range(len(self.balloons) -1, -1, -1):
+                    if self.balloons[i] == top_balloon or self.balloons[i] == second_balloon:
+                        the_biggest = self.balloons[i]
+                        self.balloons_amount[the_biggest] -= 1
```

Comment:
this is the section that is having issues. i think what is happening is it reads self.balloons[i] == top_balloon first and goes through self.balloons looking for that first then goes through the list looking for second balloon. What i was wanting was for it to go through the list and look for them both at the same time and which ever appears first becomes the biggest